src/datatype.jl

```diff
@@ -0,0 +1,36 @@
+# custom data tyoe to hold ImageHDU
+struct CCDData{T,M<:AbstractMatrix{T}} <: AbstractMatrix{T}
+    data::M
+    hdr::FITSHeader
+end
+
+# extending the AbstractMatrix interface
+Base.size(A::CCDData) = size(A.data)
+Base.getindex(A::CCDData, i) = getindex(A.data, i)
+Base.setindex!(A::CCDData, v, i) = setindex!(A.data, v, i)
+Base.eachindex(A::CCDData) = eachindex(A.data)
+Base.iterate(A::CCDData) = iterate(A.data, state=(eachindex(A),))
```

Comment:
Shouldn't this method have a `state` optional argument?

Reply:
Yes, it should have one. I will add it.

Reply:
you should be able to get away with
```julia
Base.iterate(A::CCDData) = iterate(A.data)
Base.iterate(A::CCDData, state) = iterate(A.data, state)
```